class.two-factor-core.php

```diff
@@ -318,6 +318,12 @@ public static function login_html( $user, $login_nonce, $redirect_to, $error_msg
 		$interim_login = isset( $_REQUEST['interim-login'] ); // WPCS: override ok.
 		$wp_login_url = wp_login_url();
 
+		// If we're running in a WP-Engine environment then we need to apply the
+		// site_url filter with the login_post scheme for the form action.
+		if ( function_exists( 'is_wpe' ) && is_wpe() ) {
```

Comment:
@ccoley This is a host specific integration and the function name is pretty generic and could conflict with existing code bases.

In #256 @cameronjonesweb suggested to switch to `site_url()` for the form action URLs to match how the [WP core does it](https://github.com/WordPress/WordPress/blob/da7a80d67fea29c2badfc538bfc01c8a585f0cbe/wp-login.php#L1082), which should also resolve this issue.

Reply:
@kasparsd I was originally going to use `site_url()` but I looked back in the code and saw that you were originally using that, but then changed it to use `wp_login_url()` in [this pull request](https://github.com/georgestephanis/two-factor/pull/62/files/bce8bf33e8177839d1027099f5fe33abd8e9191b#diff-fa3c61990ea53ab19ab294efedcfeffa) to fix WooCommerce login forms and issue #61.

I thought that my way of doing it was less likely to cause negative side effects, but I'm happy to use `site_url()` instead if that is what you want.

Reply:
Also, if the `is_wpe()` function is too generic, then we could also check for existence of the `IS_WPE` environment variable.

Reply:
@ccoley That is great find regarding the #62!

All of the two-factor processing currently relies on the `wp-login.php` logic [and hooks](https://github.com/georgestephanis/two-factor/blob/99d654404d2bd908c710649bfc04a37274c23b6b/class.two-factor-core.php#L41-L42) so we must `POST` the form to `wp-login.php`. I'm not sure it has ever worked with a filtered `wp-login.php` destination. Will test it with a local install of WooCommerce.

Reply:
[Confirmed](https://github.com/georgestephanis/two-factor/issues/256#issuecomment-435565107) that WooCommerce on-page login form works as expected with the `wp-login.php` form action.